api/src/main/java/org/apache/iceberg/TrackedFile.java

```diff
@@ -0,0 +1,280 @@
+   */
+  List<Integer> equalityIds();
+
+  /**
+   * Returns the location of the referenced file.
+   *
+   * <p>For POSITION_DELETES or MANIFEST_DV: location of data file that the DV references.
+   *
+   * <p>For DELETE_MANIFEST: location of affiliated data manifest, or null if unaffiliated.
+   */
+  String referencedFile();
+
+  /**
+   * Copies this tracked file.
+   *
+   * <p>Manifest readers can reuse file instances; use this method to copy data when collecting
+   * files from tasks.
+   */
+  F copy();
+
+  /**
+   * Copies this tracked file without stats.
+   *
+   * <p>Use this method to copy data without stats when collecting files.
+   */
+  F copyWithoutStats();
+
+  /**
+   * Copies this tracked file with stats only for specific columns.
+   *
+   * <p>Manifest readers can reuse file instances; use this method to copy data with stats only for
+   * specific columns when collecting files.
+   *
+   * @param requestedColumnIds column IDs for which to keep stats
+   * @return a copy of this tracked file, with content stats for only the requested columns
+   */
+  F copyWithStats(Set<Integer> requestedColumnIds);
+
+  /**
+   * Returns the ordinal position in the manifest.
+   *
+   * <p>Used for applying manifest deletion vectors.
+   */
+  Long pos();
+
+  /**
+   * Converts this tracked file to a DataFile.
+   *
+   * <p>Only valid when content_type is DATA. Extracts partition from contentStats using the
+   * provided spec.
+   *
+   * @param spec the partition spec (needed to know which transforms to apply)
+   * @return a DataFile representation
+   * @throws IllegalStateException if content_type is not DATA
+   * @throws UnsupportedOperationException if ContentStats not yet implemented
+   */
+  DataFile asDataFile(PartitionSpec spec);
+
+  /**
+   * Converts this tracked file to a DeleteFile.
+   *
+   * <p>Only valid when content_type is POSITION_DELETES or EQUALITY_DELETES. Extracts partition
+   * from contentStats using the provided spec.
+   *
+   * @param spec the partition spec (needed to know which transforms to apply)
+   * @return a DeleteFile representation
+   * @throws IllegalStateException if content_type is not a delete type
+   * @throws UnsupportedOperationException if ContentStats not yet implemented
+   */
+  DeleteFile asDeleteFile(PartitionSpec spec);
+
+  /** Set the status for this tracked file entry. */
+  void setStatus(TrackingInfo.Status status);
+
+  /** Set the snapshot ID for this tracked file entry. */
+  void setSnapshotId(Long snapshotId);
+
+  /** Set the data sequence number for this tracked file entry. */
+  void setSequenceNumber(Long sequenceNumber);
+
+  /** Set the file sequence number for this tracked file entry. */
+  void setFileSequenceNumber(Long fileSequenceNumber);
+
+  /** Set the first row ID for this tracked file entry. */
+  void setFirstRowId(Long firstRowId);
+
+  /** Set the ordinal position in the manifest. */
+  void setPos(Long position);
```

Comment:
I don't think that this is needed. In the v3 implementation, we use a column that projects _pos without calling a setter.